src/opm/io/eclipse/LODSmryOutput.cpp

```diff
@@ -0,0 +1,186 @@
+/*
+   Copyright 2019 Statoil ASA.
+
+   This file is part of the Open Porous Media project (OPM).
+
+   OPM is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   OPM is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+   */
+
+#include <opm/io/eclipse/EclUtil.hpp>
+#include <opm/io/eclipse/EclFile.hpp>
+#include <opm/io/eclipse/LODSmryOutput.hpp>
+#include <opm/io/eclipse/EclOutput.hpp>
+
+#include <opm/parser/eclipse/EclipseState/EclipseState.hpp>
+
+#include <opm/common/utility/TimeService.hpp>
+
+//#include <iostream>
+//#include <iomanip>
+#include <string>
+
+
+namespace Opm { namespace EclIO {
+
+
+LODSmryOutput::LODSmryOutput(const std::vector<std::string>& valueKeys, const std::vector<std::string>& valueUnits,
+                 const EclipseState& es, const time_t start_time)
+{
+    m_nVect = valueKeys.size();
+    m_nTimeSteps = 0;
+
+    IOConfig ioconf = es.getIOConfig();
+
+    m_restart_rootn = "";
+    m_restart_step = -1;
+
+    InitConfig initcfg = es.getInitConfig();
+
+    if (initcfg.restartRequested()) {
+        m_restart_rootn = initcfg.getRestartRootName();
+        m_restart_step = initcfg.getRestartStep();
+    }
+
+    m_fmt = es.cfg().io().getFMTOUT();
+
+    auto dims = es.gridDims();
+
+    m_outputFileName = ioconf.getOutputDir() + "/" + ioconf.getBaseName() + ".LODSMRY";
+
+    m_smry_keys = this->make_modified_keys(valueKeys, dims);
+    m_smryUnits = valueUnits;
+
+    Opm::time_point startdat = Opm::TimeService::from_time_t(start_time);
+
+    Opm::TimeStampUTC ts( std::chrono::system_clock::to_time_t( startdat ));
+
+    m_start_date_vect = {ts.day(), ts.month(), ts.year(),
+        ts.hour(), ts.minutes(), ts.seconds(), 0 };
+
+    for (size_t n = 0; n < static_cast<size_t>(m_nVect); n++)
+        m_smrydata.push_back({});
+}
+
+
+void LODSmryOutput::write(const std::vector<float>& ts_data, int report_step)
+{
+
+    if (ts_data.size() != static_cast<size_t>(m_nVect))
+        throw std::invalid_argument("size of ts_data vector not same as number of smry vectors");
```

Comment:
You should probably include `<stdexcept>` to get a declaration of `std::invalid_argument`.